scripts/gen_api_vimdoc.py

```diff
@@ -471,67 +492,100 @@ def gen_docs(config):
+
+    for filename in luadoc_section_order:
+        if filename not in sections:
+            continue
+        title, helptag, section_doc = sections.pop(filename)
+
+        i += 1
+        luadocs += sep
+        luadocs += '\n%s%s' % (title, helptag.rjust(text_width - len(title)))
+        luadocs += section_doc
+        luadocs += '\n\n\n'
 
     if sections:
         # In case new API sources are added without updating the order dict.
         for title, helptag, section_doc in sections.values():
             i += 1
-            docs += sep
-            docs += '\n%s%s' % (title, helptag.rjust(text_width - len(title)))
-            docs += section_doc
-            docs += '\n\n\n'
-
-    docs = docs.rstrip() + '\n\n'
-    docs += ' vim:tw=78:ts=8:ft=help:norl:\n'
-
-    doc_file = os.path.join(base_dir, 'runtime/doc', doc_filename)
-    delete_lines_below(doc_file, section_start_token)
-    with open(doc_file, 'ab') as fp:
-        fp.write(docs.encode('utf8'))
+            apidocs += sep
+            apidocs += '\n%s%s' % (title, helptag.rjust(text_width - len(title)))
+            apidocs += section_doc
+            apidocs += '\n\n\n'
+
+    apidocs = apidocs.rstrip() + '\n\n'
+    apidocs += ' vim:tw=78:ts=8:ft=help:norl:\n'
+    luadocs = luadocs.rstrip() + '\n\n'
+    luadocs += ' vim:tw=78:ts=8:ft=help:norl:\n'
+
+    luadoc_file = os.path.join(base_dir, 'runtime/doc', luadoc_filename)
+    apidoc_file = os.path.join(base_dir, 'runtime/doc', doc_filename)
+
+    delete_lines_below(apidoc_file, api_section_start_token)
+    with open(apidoc_file, 'ab') as fp:
+        fp.write(apidocs.encode('utf8'))
+
+    delete_lines_below(luadoc_file, luadoc_section_start_token)
+    with open(luadoc_file, 'ab') as fp:
+        fp.write(luadocs.encode('utf8'))
     shutil.rmtree(out_dir)
 
 
 def filter_source(filename):
-    """Filters the source to fix macros that confuse Doxygen."""
-    with open(filename, 'rt') as fp:
-        print(re.sub(r'^(ArrayOf|DictionaryOf)(\(.*?\))',
-                     lambda m: m.group(1)+'_'.join(
-                         re.split(r'[^\w]+', m.group(2))),
-                     fp.read(), flags=re.M))
+    name, extension = os.path.splitext(filename)
+    if extension == '.lua':
+        p = subprocess.run([lua2dox_filter, filename], stdout=subprocess.PIPE)
+        op = ('?' if 0 != p.returncode else p.stdout.decode('utf-8'))
+        print(op)
+    else:
+        """Filters the source to fix macros that confuse Doxygen."""
+        with open(filename, 'rt') as fp:
+            print(re.sub(r'^(ArrayOf|DictionaryOf)(\(.*?\))',
+                         lambda m: m.group(1)+'_'.join(
+                             re.split(r'[^\w]+', m.group(2))),
+                         fp.read(), flags=re.M))
 
 
 # Doxygen Config {{{
 Doxyfile = '''
 OUTPUT_DIRECTORY       = {output}
 INPUT                  = {input}
 INPUT_ENCODING         = UTF-8
-FILE_PATTERNS          = *.h *.c
+FILE_PATTERNS          = *.h *.c *.lua
```

Comment:
Might want to set this based on the "mode". Thus:

    FILE_PATTERNS             = "{file_patterns}"
    ...
    
then later in `gen_docs()` ...

```
    p.communicate(config.format(input=src_dir, output=out_dir,
                                file_patterns=CONFIG[mode]["file_patterns"],
                                filter=filter_cmd).encode('utf8')) 
```